tests/Functional/Schema/DefaultValueTest.php

```diff
@@ -31,6 +33,35 @@ protected function setUp(): void
         $this->connection->insert('default_value', ['id' => 1]);
     }
 
+    public function testDefaultValueAsFunction(): void
+    {
+        $platform     = $this->connection->getDatabasePlatform();
+        $defaultValue = 'uuid()';
+        if ($platform instanceof PostgreSQLPlatform) {
+            $defaultValue = 'gen_random_uuid()';
+        }
+
+        if ($platform instanceof SQLServerPlatform) {
+            $defaultValue = 'newid()';
+        }
+
+        $table = new Table('default_value');
+        $table->addColumn('id', Types::STRING, ['default' => $defaultValue, 'length' => 36]);
+
+        $this->dropAndCreateTable($table);
+
+        $sm          = $this->connection->createSchemaManager();
+        $onlineTable = $sm->introspectTable('default_value');
+
+        self::assertSame($defaultValue, $onlineTable->getColumn('id')->getDefault());
+
+        self::assertTrue(
+            $sm->createComparator()
+                ->compareTables($table, $onlineTable)
+                ->isEmpty(),
+        );
```

Comment:
This is good! Now, what happens if the introspected table…

* … has no default?
* … has a literal value as default?
* … has `NULL` as default?
* … has a different expression as default?

In that case, we would expect the schema manager to execute the proper SQL to migrate the deployed table and we need to verify that this migration performed successfully.

This has to be covered with tests as well.

Sames goes for the opposite direction: The schema deployed to the database has a column with a default expression and we want to change that to a literal, `NULL`, no default or a different expression.